query/src/main/java/io/keen/client/java/Percentile.java

```diff
@@ -0,0 +1,108 @@
+package io.keen.client.java;
+
+/**
+ * A Percentile representation enforcing the format and range specified by the Keen IO APIs.
+ *
+ * @author masojus
+ */
+public class Percentile {
+    private static final double MIN_PERCENTILE = 0.01;
+    private static final double MAX_PERCENTILE = 100.00;
+
+    private final Double percentile;
+
+    private Percentile(double percentile) {
+        this.percentile = percentile;
+    }
+
+    /**
+     * Gets the raw Double.
+     *
+     * @return The raw Double representation of this percentile.
+     */
+    public Double asDouble() {
+        return this.percentile;
+    }
+
+    /**
+     * Creates a Percentile instance with a Double value if in the valid range as described in the
+     * <a href="https://keen.io/docs/api/#percentile">Keen IO APIs</a>. This factory will throw
+     * if the given percentile does not strictly satisfy the rules in those API docs. Use
+     * {@link #createCoerced(double)} if clamping and rounding is fine.
+     *
+     * @param percentile The percentile to calculate, supporting (0, 100.00] with two decimal places
+     *                   of precision.
+     * @return A Percentile instance representing the percentile to calculate.
+     */
+    public static Percentile createStrict(double percentile) {
+        if (MIN_PERCENTILE > percentile || MAX_PERCENTILE < percentile) {
+            throw new IllegalArgumentException("The 'percentile' must be in the range (0, 100].");
+        }
+
+        if (getNumDecimalPlaces(percentile) > 2) {
```

Comment:
Comment: I actually didn't know that we limit percentiles to 2 decimal places. Two and a half years here and I still learn new things about our API ;)